### PR TITLE
Feat/show loading card

### PR DESCRIPTION
**Pull request by graykode**

### CODE REVIEW

**Review by onemonster — CHANGES_REQUESTED** [review]
Instead of using an arbitrary 10 seconds to filter out pending deposits, why don't we use the block number from the subgraph, and compare the block number of the subgraph with the block number of the transaction that created the deposit. If the block number is bigger than the subgraph we consider it pending.

File: contexts/deposit-context.tsx [hunk 1]
```diff
@@ -134,8 +165,17 @@ export const DepositProvider = ({ children }: React.PropsWithChildren<{}>) => {
           value: calculateETHValue(asset.underlying, amount),
           account: walletClient.account,
         })
+        previousConformationPositions.current = []
```
Comment (Dorvin) [inline contexts/deposit-context.tsx:168]: p2; need to remove this line

**Review by Dorvin — APPROVED** [review]

File: contexts/borrow-context.tsx [hunk 2]
```diff
@@ -92,11 +93,40 @@ export const BorrowProvider = ({ children }: React.PropsWithChildren<{}>) => {
...
+        selectedChain.id,
+        userAddress,
+      )
+      // pending positions flush conditions:
+      // 1) pending positions are not empty
+      // 2) current conformation positions is updated compared to previous conformation positions
+      if (
+        pendingPositions.length > 0 &&
+        previousConformationPositions.current !== undefined &&
+        conformationPositions.length !==
```
Comment (onemonster) [inline contexts/borrow-context.tsx:119]: Using length to check equality could cause unexpected bugs.

File: contexts/borrow-context.tsx [hunk 3]
```diff
@@ -92,11 +93,30 @@ export const BorrowProvider = ({ children }: React.PropsWithChildren<{}>) => {
...
   const { data: positions } = useQuery(
-    ['loan-positions', userAddress, selectedChain],
-    () =>
-      userAddress ? fetchLoanPositions(selectedChain.id, userAddress) : [],
+    ['loan-positions', userAddress, selectedChain, pendingPositions],
+    async () => {
+      if (!userAddress) {
+        return []
+      }
+      const conformationPositions = await fetchLoanPositions(
```
Comment (onemonster) [inline contexts/borrow-context.tsx:106]: `confirmedPositions`

**Review by Dorvin — APPROVED** [review]

**Review by onemonster — APPROVED** [review]

**Pull request merged**